### PR TITLE
Update Azure.ResourceManager.KeyVault 

### PR DESCRIPTION
**Pull request by ArthurMa1978**

### CODE REVIEW

File: sdk/keyvault/Azure.ResourceManager.KeyVault/src/autorest.md [hunk 1]
```diff
@@ -5,17 +5,20 @@ Run `dotnet build /t:GenerateCode` to generate code.
...
 clear-output-folder: true
 skip-csproj: true
 namespace: Azure.ResourceManager.KeyVault
 modelerfour:
-  lenient-model-deduplication: true
+  flatten-payloads: false
 model-namespace: false
+override-operation-name:
+  Vaults_CheckNameAvailability: CheckKeyVaultNameAvailability
+  MHSMPrivateLinkResources_ListByMHSMResource: GetMhsmPrivateLinkResourcesByMhsmResource
```
Comment (ArcturusZhang) [inline sdk/keyvault/Azure.ResourceManager.KeyVault/src/autorest.md:18]:
```suggestion
  MHSMPrivateLinkResources_ListByMhsmResource: GetMhsmPrivateLinkResourcesByMhsmResource
```

**Review by ArcturusZhang — COMMENTED** [review]
To get the 100% accurate operationId, please try enable the following configuration:
```
mgmt-debug:
  show-request-path: true
```

File: sdk/keyvault/Azure.ResourceManager.KeyVault/src/autorest.md (hunk 1 above)
Comment (ArcturusZhang) [reply sdk/keyvault/Azure.ResourceManager.KeyVault/src/autorest.md:18]: And BTW I think it makes more sense if you rename this operation to `GetPrivateLinkResources` or `GetMhsmPrivateLinkResource`

**Comment by ArthurMa1978** [comment]
close this pr as the changes have been merged into https://github.com/Azure/azure-sdk-for-net/pull/26419

**Pull request closed without merging**